### ISSUE TITLE
Missing reference to Microsoft.Azure.Services.AppAuthentication 1.1.0-preview in a NuGet package

### ISSUE THREAD

**Issue by alexsorokoletov** [Azure/azure-event-hubs-dotnet#270]
## Actual Behavior
1. Install package into a Xamarin.Android app
2. Compile app
3. Compilation fails with error:
```
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1528,2): error : Exception while loading assemblies: System.IO.FileNotFoundException: Could not load assembly 'Microsoft.Azure.Services.AppAuthentication, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Perhaps it doesn't exist in the Mono for Android profile?
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1528,2): error : File name: 'dMicrosoft.Azure.Services.AppAuthentication.ll'
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1528,2): error :   at Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference reference, Mono.Cecil.ReaderParameters parameters) [0x00099] in /Users/builder/data/lanes/5809/a7829590/source/monodroid/external/xamarin-android/external/Java.Interop/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs:229 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1528,2): error :   at Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference reference) [0x00000] in /Users/builder/data/lanes/5809/a7829590/source/monodroid/external/xamarin-android/external/Java.Interop/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs:179 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1528,2): error :   at Xamarin.Android.Tasks.ResolveAssemblies.AddAssemblyReferences (Java.Interop.Tools.Cecil.DirectoryAssemblyResolver resolver, System.Collections.Generic.ICollection`1[T] assemblies, Mono.Cecil.AssemblyDefinition assembly, System.Boolean topLevel) [0x0015c] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1528,2): error :   at Xamarin.Android.Tasks.ResolveAssemblies.Execute (Java.Interop.Tools.Cecil.DirectoryAssemblyResolver resolver) [0x001c7] in <bae064cadfb64dccb729d7cb7dab7861>:0 

```

## Expected Behavior
3. Compilation succeeds
 

## Versions  
- OS platform and version: Mac OS latest
- .NET Version: .NET Standard 2.0
- NuGet package version or commit ID:
2.0.0

Looks like package dependency to Microsoft.Azure.Services.AppAuthentication is missing.
.csproject has a package reference but the final nuget does not list this as a dependency 
https://www.nuget.org/packages/Microsoft.Azure.EventHubs/

Wondering why.

Adding reference to a package Microsoft.Azure.Services.AppAuthentication 1.1.0-preview fixes the problem

**Comment by alexsorokoletov** [comment]
Adding this reference fixes DEBUG build. In RELEASE Android has to link everything and during link process we get another error:

```
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018: The "LinkAssemblies" task failed unexpectedly.
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018: Java.Interop.Tools.Diagnostics.XamarinAndroidException: error XA2006: Could not resolve reference to 'Microsoft.IdentityModel.Clients.ActiveDirectory.ClientAssertionCertificate' (defined in assembly 'Microsoft.Azure.EventHubs, Version=2.0.0.0, Culture=neutral, PublicKeyToken=7e34167dcc6d6d8c') with scope 'Microsoft.IdentityModel.Clients.ActiveDirectory, Version=3.19.1.3001, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. When the scope is different from the defining assembly, it usually means that the type is forwarded. ---> Mono.Cecil.ResolutionException: Failed to resolve Microsoft.IdentityModel.Clients.ActiveDirectory.ClientAssertionCertificate
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Steps.MarkStep.HandleUnresolvedType (Mono.Cecil.TypeReference reference) [0x00013] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Steps.MarkStep.MarkType (Mono.Cecil.TypeReference reference) [0x0002d] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at MonoDroid.Tuner.MonoDroidMarkStep.MarkType (Mono.Cecil.TypeReference reference) [0x00000] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Steps.MarkStep.MarkField (Mono.Cecil.FieldReference reference) [0x0004a] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Steps.MarkStep.InitializeFields (Mono.Cecil.TypeDefinition type) [0x00024] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Steps.MarkStep.InitializeType (Mono.Cecil.TypeDefinition type) [0x0005d] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Steps.MarkStep.InitializeAssembly (Mono.Cecil.AssemblyDefinition assembly) [0x0002e] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Steps.MarkStep.Initialize () [0x00014] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Steps.MarkStep.Process (Mono.Linker.LinkContext context) [0x00007] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Mono.Linker.Pipeline.Process (Mono.Linker.LinkContext context) [0x0001b] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at MonoDroid.Tuner.Linker.Run (Mono.Linker.Pipeline pipeline, Mono.Linker.LinkContext context) [0x00000] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at MonoDroid.Tuner.Linker.Process (MonoDroid.Tuner.LinkerOptions options, Mono.Linker.ILogger logger, Mono.Linker.LinkContext& context) [0x00071] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Xamarin.Android.Tasks.LinkAssemblies.Execute (Java.Interop.Tools.Cecil.DirectoryAssemblyResolver res) [0x002c8] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:    --- End of inner exception stack trace ---
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Java.Interop.Tools.Diagnostics.Diagnostic.Error (System.Int32 code, System.Exception innerException, System.String message, System.Object[] args) [0x00000] in /Users/builder/data/lanes/5809/a7829590/source/monodroid/external/xamarin-android/external/Java.Interop/src/Java.Interop.Tools.Diagnostics/Java.Interop.Tools.Diagnostics/Diagnostic.cs:165 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Xamarin.Android.Tasks.LinkAssemblies.Execute (Java.Interop.Tools.Cecil.DirectoryAssemblyResolver res) [0x0046f] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Xamarin.Android.Tasks.LinkAssemblies.Execute () [0x001fa] in <bae064cadfb64dccb729d7cb7dab7861>:0 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute () [0x00023] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/msbuild-15/src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs:631 
    /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1748,5): error MSB4018:   at Microsoft.Build.BackEnd.TaskBuilder+<ExecuteInstantiatedTask>d__26.MoveNext () [0x00212] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/msbuild-15/src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs:787 
  Done executing task "LinkAssemblies" -- FAILED.
Done building target "_LinkAssembliesShrink" in project "***.Android.csproj" -- FAILED.
```

Found a bit similar (but not exactly) issue here https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/511

**Comment by serkantkaraca** [comment]
Thanks for reporting the issue. We will work on this and update the thread.

**Comment by serkantkaraca** [comment]
It seems master and dev branch dependencies are different. We will address the issue in our next release which will happen in the next couple days.

**Comment by alexsorokoletov** [comment]
Thank you, Serkant. 
Can you please also check if you can build an Android app with event hub sdk in the Release configuration?



Alex

**Comment by serkantkaraca** [comment]
I think that will also fail, no need to check. I have bumped up the AppAuth package reference to 1.0.0, seems they have released this 2 days ago. You can unblock your development by using packages from here until we have new SDK: https://ci.appveyor.com/project/serkantkaraca/azure-event-hubs-dotnet/branch/dev/job/t2pb0uk2kipw6vjm/artifacts

**Comment by serkantkaraca** [comment]
Released 2.1
